File: 1-binary.js
const assert = (condition, error) => {
  if (!condition) {
    throw new Error(error)
  }
}

function solution (number) {
  const binary = number.toString(2)

  let isGap = false
  let current = 0
  let longest = 0

  for (let i = 0; i < binary.length; i++) {
    if (binary[i] === '1') {
      if (isGap) {
        if (current > longest) {
          longest = current
        }

        current = 0
      } else {
        isGap = true
      }
    } else {
      current++
    }
  }

  return longest
}

assert(solution(parseInt('0', 2)) === 0, '0 returns 0 ')
assert(solution(parseInt('1111', 2)) === 0, '1111 returns 0 ')
assert(solution(parseInt('111111101111', 2)) === 1, '111111101111 returns 1 ')
assert(solution(parseInt('0100001', 2)) === 4, '0100001 returns 4')
assert(solution(parseInt('101000', 2)) === 1, '101000 returns 1')
assert(solution(parseInt('101001000', 2)) === 2, '101001000 returns 2')
assert(solution(parseInt('10110001', 2)) === 3, '10110001 returns 2')
assert(solution(parseInt('10000000000000000000000000001', 2)) === 27, '10000000000000000000000000001 return 27')

File: 1-binary.py
def solution(N):
  binary = bin(N)[2:]

  in_gap = False
  longest = 0
  current = 0

  for bit in binary:
    if bit == '1':
      if in_gap:
        if current > longest:
          longest = current

        current = 0
      else:
        in_gap = True
    else:
      current += 1

  return longest

assert solution(0) == 0
assert solution(0b1111) == 0
assert solution(0b111111101111) == 1
assert solution(0b0100001) == 4
assert solution(0b101000) == 1
assert solution(0b101001000) == 2
assert solution(0b10110001) == 3
assert solution(0b10000000000000000000000000001) == 27
